mlflow/gateway/config.py

```diff
@@ -240,12 +240,24 @@ def to_route(self) -> "Route":
         )
 
 
-class Route(BaseModel, extra=Extra.allow):
```

Comment:
`extra=Extra.allow` is incompatible with `Config`. Looking into https://docs.pydantic.dev/1.10/usage/model_config/#extra-attributes more, I think we should be using `extra=Extra.ignore` anyway, else attributes from updated backends will start to surface in old clients, potentially causing confusion

Reply:
cc @BenWilson2 @harupy I'll file another PR to change to `Extra.ignore`

Reply:
Filed https://github.com/mlflow/mlflow/pull/8940 to address